### PR TITLE
Add additional detail and clarifications on SELECT

### PR DESCRIPTION
**Pull request by bitwombat**
- Also the effect of WHERE on result array.

As per #6563

### CODE REVIEW

File: docs/en/reference/dql-doctrine-query-language.rst [hunk 1]
```diff
@@ -49,9 +49,12 @@ SELECT queries
 DQL SELECT clause
 ~~~~~~~~~~~~~~~~~
 
-The select clause of a DQL query specifies what appears in the
-query result. The composition of all the expressions in the select
-clause also influences the nature of the query result.
+The SELECT clause of a DQL query specifies what gets hydrated in
+the query result. You are always returned usable objects, but any
+associated objects not included in the SELECT clause will be
+proxies (ie. unhydrated).  They get hydrated by Doctrine when
```
Comment (Ocramius) [inline docs/en/reference/dql-doctrine-query-language.rst]: Too much information here - this should be detailed somewhere else

File: docs/en/reference/dql-doctrine-query-language.rst [hunk 2]
```diff
@@ -83,14 +86,48 @@ Lets examine the query:
...
+The composition of the expressions in the SELECT clause also
+influences the nature of the query result. There are three
+cases:
+
+- All objects. For example:
+    ``SELECT u, p, n FROM Users u...``
+  In this case, the result array will be made up of objects of
+  the type in the FROM clause.  In the example above, the query
+  will return an array of User objects, with associated classes
+  identify else where in the query as 'p' and 'n' hydrated.
```
Comment (Ocramius) [inline docs/en/reference/dql-doctrine-query-language.rst]: The wording is a bit confusing here - can you rephrase it so that it is clear that `p` and `n` are children of `u`, and are not going to be selected as separate columns in the result?

File: docs/en/reference/dql-doctrine-query-language.rst [hunk 3]
```diff
@@ -83,14 +86,48 @@ Lets examine the query:
...
+
+  You can select scalars from any entity in the query. 
+
+- Mixed.  For example:
+    ``SELECT u, p.quantity FROM Users u...``
+  Here, the result will again be an array of arrays, with each
+  element being an array made up of a User object and the scalar
+  value p.quantity.
+
+Multiple FROM clauses are allowed, which would cause the result
```
Comment (Ocramius) [inline docs/en/reference/dql-doctrine-query-language.rst]: Use double backticks around anything that is code-like, such as `FROM`

File: docs/en/reference/dql-doctrine-query-language.rst [hunk 4]
```diff
@@ -83,14 +86,48 @@ Lets examine the query:
...
+  example above, each element of the result array would be an
+  array of the scalar name and address values. 
+
+  You can select scalars from any entity in the query. 
+
+- Mixed.  For example:
+    ``SELECT u, p.quantity FROM Users u...``
+  Here, the result will again be an array of arrays, with each
+  element being an array made up of a User object and the scalar
+  value p.quantity.
```
Comment (Ocramius) [inline docs/en/reference/dql-doctrine-query-language.rst]: Use double backticks around anything that is code-like, such as `p.quantity`

File: docs/en/reference/dql-doctrine-query-language.rst (hunk 2 above)
Comment (Ocramius) [inline docs/en/reference/dql-doctrine-query-language.rst]: Use double backticks around anything that is code-like, such as `u`, `p`, `n`

File: docs/en/reference/dql-doctrine-query-language.rst [hunk 5]
```diff
@@ -83,14 +86,48 @@ Lets examine the query:
...
+
+Multiple FROM clauses are allowed, which would cause the result
+array elements to cycle through the classes included in the
+multiple FROM clauses.
+
+.. note::
+
+    You cannot select other entities unless you also select the
```
Comment (Ocramius) [inline docs/en/reference/dql-doctrine-query-language.rst]: This case only applies to the scenario where entities are being selected, not the scalars scenario

File: docs/en/reference/dql-doctrine-query-language.rst [hunk 6]
```diff
@@ -83,14 +86,48 @@ Lets examine the query:
...
+    root of the selection (the first entity in FROM).
+
+    Doctrine tells you you have violated this constraint with the
+    exception "Cannot select entity through identification
```
Comment (Ocramius) [inline docs/en/reference/dql-doctrine-query-language.rst]: "with an exception" - the exception message should be improved in the library if it isn't clear enough there.

I would also add an example of when this would fail (example query)

File: docs/en/reference/dql-doctrine-query-language.rst [hunk 7]
```diff
@@ -83,14 +86,48 @@ Lets examine the query:
...
+    variables without choosing at least one root entity alias."
+
+It is possible to wrap both fields and identification values into
```
Comment (Ocramius) [inline docs/en/reference/dql-doctrine-query-language.rst]: This block is confusing - why is it in this section?

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: docs/en/reference/dql-doctrine-query-language.rst (hunk 1 above)
Comment (bitwombat) [inline docs/en/reference/dql-doctrine-query-language.rst]: Somewhere else in the same doc?

File: docs/en/reference/dql-doctrine-query-language.rst (hunk 2 above)
Comment (bitwombat) [inline docs/en/reference/dql-doctrine-query-language.rst]: Wow, typo on my part.  Bad news.  OK, will fix, rephrase and add backticks.

File: docs/en/reference/dql-doctrine-query-language.rst (hunk 5 above)
Comment (bitwombat) [inline docs/en/reference/dql-doctrine-query-language.rst]: It's a note, not coupled to the scalars scenario.  Unless I'm not seeing something.

File: docs/en/reference/dql-doctrine-query-language.rst (hunk 1 above)
Comment (Ocramius) [inline docs/en/reference/dql-doctrine-query-language.rst]: Possibly, but remember that you are approaching readers that may have no clue about doctrine, so this stuff can be plain confusing until the basics are covered :-)

Comment (Ocramius) [inline docs/en/reference/dql-doctrine-query-language.rst]: Basically, ELI5

File: docs/en/reference/dql-doctrine-query-language.rst (hunk 6 above)
Comment (bitwombat) [inline docs/en/reference/dql-doctrine-query-language.rst]: The message does call the same thing an "identification variable" and an "alias" in the same message.  As do the docs.  I'm wondering if one term should be picked and used throughout.

File: docs/en/reference/dql-doctrine-query-language.rst (hunk 5 above)
Comment (Ocramius) [inline docs/en/reference/dql-doctrine-query-language.rst]: Alright then :+1:

File: docs/en/reference/dql-doctrine-query-language.rst (hunk 7 above)
Comment (bitwombat) [inline docs/en/reference/dql-doctrine-query-language.rst]: I think you might have selected the wrong lines there, as that's not a block... ??

**Comment by Ocramius** [comment]
One note on positioning knowledge in the document: after editing, try to scan the entire document again and see if it makes sense or it actually asks the user to jump around "goto" style :-)

File: docs/en/reference/dql-doctrine-query-language.rst (hunk 7 above)
Comment (Ocramius) [inline docs/en/reference/dql-doctrine-query-language.rst]: Sorry, miswording on my side - I mean paragraph. The information about functions being applicable to the selection seems weird if it's redundant with a later section. Maybe move this information directly there?

File: docs/en/reference/dql-doctrine-query-language.rst (hunk 6 above)
Comment (Ocramius) [inline docs/en/reference/dql-doctrine-query-language.rst]: > I'm wondering if one term should be picked and used throughout.

Possibly - it would surely reduce impedance between ORM internals and their documentation (as they are called "identification variable" internally too)

File: docs/en/reference/dql-doctrine-query-language.rst (hunk 7 above)
Comment (bitwombat) [inline docs/en/reference/dql-doctrine-query-language.rst]: Line 127 above?  That was in the original (unchanged by me).

File: docs/en/reference/dql-doctrine-query-language.rst (hunk 1 above)
Comment (bitwombat) [inline docs/en/reference/dql-doctrine-query-language.rst]: Actually, near line 181 you already have a description of the lazy-loading and walking the object tree... so you've either changed your mind or I'm duplicating things.  :)

Either way I should delete this part.  Except I'm not sure it's clear that including things in the SELECT clause is what hydrates them.  The discussion of fetch joins at the top is attempting to do that.

What do you think?

Comment (Ocramius) [inline docs/en/reference/dql-doctrine-query-language.rst]: > Either way I should delete this part. Except I'm not sure it's clear that including things in the SELECT clause is what hydrates them. The discussion of fetch joins at the top is attempting to do that.

As long as it's documented somewhere else, heh.

File: docs/en/reference/dql-doctrine-query-language.rst (hunk 3 above)
Comment (bitwombat) [inline docs/en/reference/dql-doctrine-query-language.rst]: I'm following convention.  Elsewhere in the doc, the DQL keywords (SELECT, FROM, WITH) are not surrounded by backticks (though entire DQL commands are).  Maybe Previous You thought that would clutter the output?  Tell me what you want me to do here.

File: docs/en/reference/dql-doctrine-query-language.rst (hunk 7 above)
Comment (bitwombat) [inline docs/en/reference/dql-doctrine-query-language.rst]: Didn't hear back from you on this.  Leaving it in there since it's yours originally.

Comment (Ocramius) [inline docs/en/reference/dql-doctrine-query-language.rst]: Alright :+1:

File: docs/en/reference/dql-doctrine-query-language.rst (hunk 3 above)
Comment (Ocramius) [inline docs/en/reference/dql-doctrine-query-language.rst]: I wrote almost none of the docs, but the separate rendering format of code as code aids readability

Comment (bitwombat) [inline docs/en/reference/dql-doctrine-query-language.rst]: So you'd like me to surround all the exist SELECT, FROM, WITH, etc. statements with backticks as part of this PR?  No prob, just clarifying.

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
:ship: 

Thanks @bitwombat!
